### DISCUSSION TITLE
[CHORE] Elsa 3.1.3 Create custom role activity code?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5791] [Q&A]
Elsa 3.1.3 Create custom role activity code?

**Comment by glime-ai** [comment]
To create a custom role activity in Elsa 3.1.3, you would need to follow the general steps for creating custom activities in Elsa, as outlined in the Elsa documentation. Here's a simplified example of how you might create a custom activity that checks if a user has a specific role:

1. Create a class that inherits from `CodeActivity` or another appropriate base class if you need more complex behavior.
2. Define any inputs, outputs, or properties that your activity requires.
3. Implement the `Execute` method to contain the logic for your role check.
4. Register your custom activity with the Elsa workflow runtime.

Here's a basic example in C#:

```csharp
using Elsa.ActivityResults;
using Elsa.Attributes;
using Elsa.Services;
using Elsa.Services.Models;

namespace MyElsaExtensions.Activities
{
    [Activity(
        Category = "User",
        DisplayName = "Check User Role",
        Description = "Checks if the user has the specified role",
        Outcomes = new[] { "True", "False" }
    )]
    public class CheckUserRole : CodeActivity
    {
        [ActivityInput(Hint = "Enter the role to check.")]
        public string Role { get; set; }

        protected override IActivityExecutionResult OnExecute(ActivityExecutionContext context)
        {
            // Replace this with your actual role-checking logic.
            bool hasRole = CheckUserRoleLogic(context.User, Role);

            // Return an outcome based on whether the user has the role.
            return Outcome(hasRole ? "True" : "False");
        }

        private bool CheckUserRoleLogic(IUser user, string role)
        {
            // Implement your role-checking logic here.
            // This could involve checking against a database or an identity management system.
            return user.Roles.Contains(role);
        }
    }
}
```

To register this activity, you would add it to the Elsa services configuration in your `Startup.cs` or `Program.cs`:

```csharp
services.AddElsa(elsa => elsa.AddActivity<CheckUserRole>());
```

Please note that the `CheckUserRoleLogic` method is a placeholder where you would implement the actual logic to check if a user has a role. This might involve interacting with a user management system or database.

Remember to replace the placeholder logic with your actual role-checking implementation and adjust the activity's properties and outcomes as needed for your specific use case.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
